controllers/storageclassrequest/storageclassrequest_controller_test.go

```diff
@@ -0,0 +1,493 @@
+			expectedPoolName: "test-blockpool",
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-blockpool",
+						Namespace: "test-ns",
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  fakeStorageProfile.GetSpecHash(),
+						},
+					},
+				},
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-blockpool2",
+						Namespace: "test-ns",
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: "wrongConsumer",
+							controllers.StorageProfileSpecLabel:  "0123456789",
+						},
+					},
+				},
+			},
+		},
+		{
+			label: "Valid CephBlockPool only exists for different consumer/profile",
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-blockpool",
+						Namespace: "test-ns",
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: "wrongConsumer",
+							controllers.StorageProfileSpecLabel:  "0123456789",
+						},
+					},
+				},
+			},
+		},
+		{
+			label:           "More than one valid CephBlockPool exists",
+			failureExpected: true,
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-blockpool",
+						Namespace: "test-ns",
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  fakeStorageProfile.GetSpecHash(),
+						},
+					},
+				},
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-blockpool2",
+						Namespace: "test-ns",
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  fakeStorageProfile.GetSpecHash(),
+						},
+					},
+				},
+			},
+		},
+		{
+			label:            "Request status already has valid CephResource",
+			expectedPoolName: "test-blockpool",
+			cephResources: []*v1alpha1.CephResourcesSpec{
+				{
+					Name: "test-blockpool",
+					Kind: "CephBlockPool",
+				},
+			},
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-blockpool",
+						Namespace: "test-ns",
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  fakeStorageProfile.GetSpecHash(),
+						},
+					},
+				},
+			},
+		},
+		{
+			label:            "Request status has CephResource that doesn't exist",
+			expectedPoolName: "test-blockpool",
+			cephResources: []*v1alpha1.CephResourcesSpec{
+				{
+					Name: "test-blockpool",
+					Kind: "CephBlockPool",
+				},
+			},
+		},
```

Comment:
- when will this condition arise? I'm a but confused as we don't create objects but a pool exist
- is it something like, a failed cleanup?
- same for `TestCephFsSubVolGroup`